studio.ps1

```diff
@@ -67,10 +67,24 @@ function Stage-Three {
     $kitArgs = "/q"
 
     if ($KITTYPE -eq "HLK") {
-        if (Test-Path -Path "$PSScriptRoot\Kits\HLK${HLKKITVER}\HLKSetup.exe") {
-            $kitPath = "$PSScriptRoot\Kits\HLK${HLKKITVER}\HLKSetup.exe"
-        } else {
-            $kitPath = "$PSScriptRoot\Kits\HLK${HLKKITVER}Setup.exe"
+        $mountedDisks = Get-WmiObject -Class Win32_LogicalDisk | Where-Object { $_.DriveType -eq 5 }
+        $foundHCK = $false
+
+        foreach ($disk in $mountedDisks) {
+            $hckSetupPath = Join-Path -Path $disk.DeviceID -ChildPath "HLKSetup.exe"
+            Write-Output "Checking $hckSetupPath..."
+            if (Test-Path -Path $hckSetupPath) {
+                $kitPath = $hckSetupPath
+                $foundHCK = $true
+                break
+            }
+        }
+        if (-not $foundHCK) {

```

Comment:
Initialize `$kitPath` with `$null` and check if `$kitPath` is not `$null` instead of having `$foundHCK`, a distinct variable for its existence.